It's worth noting that this fails the existing test suite. 

```
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1529926266.6214693-137085483742155/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_e53r2y2w/ansible_module_ecs_taskdefinition.py", line 488, in <module>
    main()
  File "/tmp/ansible_e53r2y2w/ansible_module_ecs_taskdefinition.py", line 456, in main
    module.params['memory'])
  File "/tmp/ansible_e53r2y2w/ansible_module_ecs_taskdefinition.py", line 246, in register_task
    response = self.ecs.register_task_definition(**params)
  File "/usr/lib/python3.6/site-packages/botocore/client.py", line 312, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/usr/lib/python3.6/site-packages/botocore/client.py", line 579, in _make_api_call
    api_params, operation_model, context=request_context)
  File "/usr/lib/python3.6/site-packages/botocore/client.py", line 634, in _convert_to_request_dict
    api_params, operation_model)
  File "/usr/lib/python3.6/site-packages/botocore/validate.py", line 291, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Unknown parameter in input: "executionRoleArn", must be one of: family, taskRoleArn, networkMode, containerDefinitions, volumes, placementConstraints

fatal: [localhost]: FAILED! => {
    "changed": false,
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_e53r2y2w/ansible_module_ecs_taskdefinition.py\", line 488, in <module>\n    main()\n  File \"/tmp/ansible_e53r2y2w/ansible_module_ecs_taskdefinition.py\", line 456, in main\n    module.params['memory'])\n  File \"/tmp/ansible_e53r2y2w/ansible_module_ecs_taskdefinition.py\", line 246, in register_task\n    response = self.ecs.register_task_definition(**params)\n  File \"/usr/lib/python3.6/site-packages/botocore/client.py\", line 312, in _api_call\n    return self._make_api_call(operation_name, kwargs)\n  File \"/usr/lib/python3.6/site-packages/botocore/client.py\", line 579, in _make_api_call\n    api_params, operation_model, context=request_context)\n  File \"/usr/lib/python3.6/site-packages/botocore/client.py\", line 634, in _convert_to_request_dict\n    api_params, operation_model)\n  File \"/usr/lib/python3.6/site-packages/botocore/validate.py\", line 291, in serialize_to_request\n    raise ParamValidationError(report=report.generate_report())\nbotocore.exceptions.ParamValidationError: Parameter validation failed:\nUnknown parameter in input: \"executionRoleArn\", must be one of: family, taskRoleArn, networkMode, containerDefinitions, volumes, placementConstraints\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE",
    "rc": 1
}
```

--- a/lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py
+++ b/lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py
@@ -323,7 +323,7 @@ def main():
 
     if module.params['execution_role_arn']:
         if not module.botocore_at_least('1.10.44'):
-            module.fail_jason(msg='botocore needs to be version 1.10.44 or higher to use task_role_arn')
+            module.fail_json(msg='botocore needs to be version 1.10.44 or higher to use execution_role_arn')
 
     for container in module.params.get('containers', []):
         for environment in container.get('environment', []):

--- a/lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py
+++ b/lib/ansible/modules/cloud/amazon/ecs_taskdefinition.py
@@ -321,7 +321,7 @@ def main():
         if not module.botocore_at_least('1.8.4'):
             module.fail_json(msg='botocore needs to be version 1.8.4 or higher to use launch_type')
 
-    if module.params['task_role_arn']:
+    if module.params['execution_role_arn']:
         if not module.botocore_at_least('1.10.44'):
             module.fail_jason(msg='botocore needs to be version 1.10.44 or higher to use task_role_arn')
 

--- a/test/integration/targets/ecs_cluster/playbooks/network_fail.yml
+++ b/test/integration/targets/ecs_cluster/playbooks/network_fail.yml
@@ -44,6 +44,26 @@
             <<: *aws_connection_info
           register: ecs_taskdefinition_creation_vpc
 
+        - name: create ecs_taskdefinition and execution_role_arn (expected to fail)
+          ecs_taskdefinition:
+            containers:
+              - name: my_container
+                image: ubuntu
+                memory: 128
+            family: "{{ resource_prefix }}-vpc"
+            execution_role_arn: not_a_real_arn
+            state: present
+            network_mode: awsvpc
+            <<: *aws_connection_info
+          ignore_errors: yes
+          register: ecs_taskdefinition_arn
+
+        - name: check that graceful failure message is returned from ecs_taskdefinition_arn
+          assert:
+            that:
+              - ecs_taskdefinition_arn.failed
+              - 'ecs_taskdefinition_arn.msg == "botocore needs to be version 1.10.44 or higher to use execution_role_arn"'
+
         - name: ecs_taskdefinition works fine even when older botocore is used
           assert:
             that:

